test/com/google/enterprise/adaptor/database/UniqueKeyTest.java

```diff
@@ -275,53 +241,23 @@ public void testProcessingDocId() throws SQLException {
         ResultSet rs = stmt.executeQuery("select * from data")) {
       UniqueKey uk = newUniqueKey("numnum:int,strstr:string");
       assertTrue("ResultSet is empty", rs.next());
-      assertEquals("345/abc", uk.makeUniqueId(rs, /*encode=*/ true));
-    }
-  }
-
-  @Test
-  public void testProcessingDocId_encodingError() throws SQLException {
-    executeUpdate("create table data(numnum int, strstr varchar)");
-    executeUpdate("insert into data(numnum, strstr) values(345, 'abc')");
-
-    try (Statement stmt = getConnection().createStatement();
-        ResultSet rs = stmt.executeQuery("select * from data")) {
-      UniqueKey uk = newUniqueKey("numnum:int,strstr:string");
-      assertTrue("ResultSet is empty", rs.next());
-      thrown.handleAssertionErrors();
-      thrown.expect(AssertionError.class);
-      thrown.expectMessage("not encoding implies exactly one parameter");
-      uk.makeUniqueId(rs, /*encode=*/ false);
+      assertEquals("345/abc", uk.makeUniqueId(rs));
     }
   }
 
   @Test
   public void testProcessingDocId_allTypes() throws SQLException {
     executeUpdate("create table data(c1 integer, c2 bigint, c3 varchar, "
         + "c4 date, c5 time, c6 timestamp)");
-    String sql = "insert into data(c1, c2, c3, c4, c5, c6) "
-        + "values (?, ?, ?, ?, ?, ?)";
-    Calendar gmt = Calendar.getInstance(TimeZone.getTimeZone("GMT"), US);
-    try (PreparedStatement ps = getConnection().prepareStatement(sql)) {
-      ps.setInt(1, 123);
-      ps.setLong(2, 4567890L);
-      ps.setString(3, "foo");
-      ps.setDate(4, new Date(0L), gmt);
-      ps.setTime(5, new Time(0L), gmt);
-      // TODO(bmj): setTimestamp with Calendar broken in H2?
-      ps.setTimestamp(6, new Timestamp(0L)/*, gmt*/);
-      assertEquals(1, ps.executeUpdate());
-    }
+    executeUpdate("insert into data(c1, c2, c3, c4, c5, c6) "
+        + "values (123, 4567890, 'foo', "
+        + "{d '2007-08-09'}, {t '12:34:56'}, {ts '2007-08-09T12:34:56'})");
 
     UniqueKey uk = new UniqueKey.Builder(
         "c1:int, c2:long, c3:string, c4:date, c5:time, c6:timestamp").build();
-
-    try (Statement stmt = getConnection().createStatement();
-        ResultSet rs = stmt.executeQuery("select * from data")) {
-      assertTrue("ResultSet is empty", rs.next());
-      assertEquals("123/4567890/foo/1970-01-01/00:00:00/0",
-          uk.makeUniqueId(rs, /*encode=*/ true));
-    }
+    ResultSet rs = executeQueryAndNext("select * from data");
+    assertEquals("123/4567890/foo/2007-08-09/12:34:56/1186688096000",
```

Comment:
This is broken, because the milliseconds depend on your local time zone. That's why I suggested using Timestamp.valueOf("2007-08-09 12:34:56").getTime()  for the expected timestamp value.